src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -0,0 +1,62 @@
+#ifndef ALPAKAWORKDIVHELPER_H
+#define ALPAKAWORKDIVHELPER_H
+
+#include "AlpakaCore/alpakaConfig.h"
+
+using namespace alpaka_common;
+
+namespace cms {
+  namespace alpakatools {
+
+    /*
+     * Creates the accelerator-dependent workdiv.
+     */
+    template <typename T_Dim>
+    WorkDiv<T_Dim> make_workdiv(const Vec<T_Dim>& blocksPerGrid, const Vec<T_Dim>& threadsPerBlockOrElementsPerThread) {
+      // On the GPU:
+      // threadsPerBlockOrElementsPerThread is the number of threads per block.
+      // Each thread is looking at a single element: elementsPerThread is always 1.
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+      const Vec<T_Dim>& elementsPerThread = Vec<T_Dim>::ones();
+      return WorkDiv<T_Dim>(blocksPerGrid, threadsPerBlockOrElementsPerThread, elementsPerThread);
```

Comment:
Hmm this allows to know what Vec<T_Dim>::ones() represent.

In general, I am not a fan of call sites like `myFunction(1024, true)`, because I think it makes it more painful for the reader to see what the `1024` or the `true` actually are.
Hence, on call site, I tend to alias the 'magic parameters values' by names, which make them directly understandable by the reader.
I would find the following much more readable:
```
const int squaredMatrixSideSize = 1024;
isSingular = true;
myFunction(quaredMatrixSideSize, isSingular);
```

It might look a bit overkill, but in practice, I find it moving away complexity, hence allowing faster reading, and moving the focus to the parts which should actually deserve it.

Here for example, with `WorkDiv<T_Dim>(blocksPerGrid, threadsPerBlockOrElementsPerThread, Vec<T_Dim>::ones()`, I would have to stop reading for an instant to try to understand what the `Vec<T_Dim>::ones() `represent, and possibly even go have a glance at the WorkDiv constructor in the library.

But ok just a matter of preference :)

Reply:
> But ok just a matter of preference :)

Indeed, which is why I'm going to merge this PR as it is.

For the callers of `make_workdiv()` I personally find the "boilerplate" harder to read for a relatively simple function signature that gets repeated often (and thus has a good chance of getting memorized). For complicated function signatures or rarely used functions I fully agree the added documentation to be helpful.

Obviously ideas for how to improve the interface of `make_workdiv()` such that the intention of the two arguments would be clear would be welcome.